src/components/CheckboxInput/CheckboxInput.test.tsx

```diff
@@ -0,0 +1,29 @@
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+import { CheckboxInput } from './CheckboxInput';
+
+describe('indeterminancy', () => {
+  it('is indeterminate when checked is "indeterminate"', () => {
+    render(<CheckboxInput checked="indeterminate" id="test" readOnly />);
+    const checkbox = screen.getByRole<HTMLInputElement>('checkbox');
+    expect(checkbox.indeterminate).toEqual(true);
+  });
+
+  it('is not indeterminate when checked is false', () => {
+    render(<CheckboxInput checked={false} id="test" readOnly />);
+    const checkbox = screen.getByRole<HTMLInputElement>('checkbox');
+    expect(checkbox.indeterminate).toEqual(false);
+  });
+
+  it('is not indeterminate when checked is true', () => {
+    render(<CheckboxInput checked id="test" readOnly />);
+    const checkbox = screen.getByRole<HTMLInputElement>('checkbox');
+    expect(checkbox.indeterminate).toEqual(false);
+  });
+
+  it('is not indeterminate when checked is uncontrolled', () => {
+    render(<CheckboxInput id="test" />);
+    const checkbox = screen.getByRole<HTMLInputElement>('checkbox');
```

Comment:
check for understanding: is a component controlled if props are passed in? I thought it was only controlled if state is managed (this excludes props in my understanding)

Reply:
> is a component controlled if props are passed in? I thought it was only controlled if state is managed

Kind of both.

It's controlled if we supply the `value`/`checked`. But doing so probably implies app developers managing the state (or at least someone managing it somewhere).